src/core/server/config/ensure_valid_configuration.ts

```diff
@@ -6,20 +6,13 @@
  * Side Public License, v 1.
  */
 
-import { getUnusedConfigKeys } from './get_unused_config_keys';
-import { ConfigService } from '../../config';
-import { CriticalError } from '../../errors';
-import { LegacyServiceSetupConfig } from '../types';
+import { ConfigService } from '@kbn/config';
+import { CriticalError } from '../errors';
 
-export async function ensureValidConfiguration(
-  configService: ConfigService,
-  { legacyConfig, settings }: LegacyServiceSetupConfig
-) {
-  const unusedConfigKeys = await getUnusedConfigKeys({
-    coreHandledConfigPaths: await configService.getUsedPaths(),
-    settings,
-    legacyConfig,
-  });
+export async function ensureValidConfiguration(configService: ConfigService) {
+  await configService.validate();
+
+  const unusedConfigKeys = await configService.getUnusedPaths();
```

Comment:
We no longer need to use combined logic between core's schema and the legacy config handled keys, but we still need to make sure that the configuration does not contain unknown keys (`configService.validate` only validates the registered schemas).